versions/1.0/SPEC.md

```diff
@@ -1036,6 +1037,42 @@ output {
 }
 ```
 
+#### Files and Optional Outputs
+
+File outputs are described as string paths relative to the initial working
+directory.
+
+```
+File somefile = "my/path/to/something.txt"
+```
+
+All file outputs are required to exist, otherwise the task will fail.
+
+However, outputs may be annotated as `Optional` types (including array types),
+in which case they will be null if files do not exist.
```

Comment:
Since there is a literal for null nowadays, it maybe good to use that here.
```suggestion
in which case they will be `None` if files do not exist.
```